src/core_plugins/kibana/server/lib/manage_uuid.js

```diff
@@ -0,0 +1,73 @@
+import uuid from 'node-uuid';
+import Promise from 'bluebird';
+import { hostname } from 'os';
+import { join as pathJoin } from 'path';
+import { readFile as readFileCallback, writeFile as writeFileCallback } from 'fs';
+
+const serverHostname = hostname();
+const FILE_ENCODING = 'utf8';
+
+export default async function manageUuid(server) {
+  const config = server.config();
+  const serverPort = server.info.port;
+  const fileName = `${serverHostname}:${serverPort}`;
+  const uuidFile = pathJoin(config.get('path.data'), fileName);
```

Comment:
is there a case where we'll need multiple uuids?  it might simplify things to have a static filename like `server_uuid`.  it'll make the filename understandable, and on the chance port is changed do we actually want a new uuid?